pkg/controller/lifecycle/actuator.go

```diff
@@ -153,18 +155,29 @@ func (a *actuator) Reconcile(ctx context.Context, logger logr.Logger, ex *extens
 		image.Tag = ptr.To[string](version.Get().GitVersion)
 	}
 
-	lakomConfig, err := yaml.JSONToYAML(a.serviceConfig.CosignPublicKeys.Raw)
+	gardenerPublicKeys, err := yaml.JSONToYAML(a.serviceConfig.CosignPublicKeys.Raw)
 	if err != nil {
 		return fmt.Errorf("failed to convert lakom config from json to yaml, %w", err)
 	}
 
+        var clientPublicKeys []byte
+	if lakomProviderConfig.PublicKeysSecretReference != nil {
+                var err error
+                clientPublicKeys, err = getClientKeys(ctx, a.client, cluster.Shoot.Spec.Resources, *lakomProviderConfig.PublicKeysSecretReference, namespace)
+                if err != nil {
+                    return fmt.Errorf("failed to find client public keys from the given reference %s", *lakomProviderConfig.PublicKeysSecretReference)
```

Comment:
```suggestion
                    return fmt.Errorf("failed to get the additional keys: %w", err)
```

It is important to wrap the original error as it will provide more information what actually failed down the stack, logging the resource reference name is kind of obsolete as it is well known and always can be read from the shoot resource.

Reply:
Absolutely, good catch.